src/rulesets/Base/nondiff.jl

```diff
@@ -22,6 +22,22 @@
 @non_differentiable ==(::Any, ::Any)
 @non_differentiable ===(::Any, ::Any)
 
+@non_differentiable accumulate(::Any, ::BitArray)
+@non_differentiable accumulate!(::Any, ::Any, ::BitArray)
+@non_differentiable Array(::BitArray)
+@non_differentiable conj(::BitArray)
+@non_differentiable cumprod(::BitArray)
+@non_differentiable cumprod!(::Any, ::BitArray)
+@non_differentiable cumsum(::BitArray)
+@non_differentiable cumsum!(::Any, ::BitArray)
+@non_differentiable DenseMatrix(::BitArray)
```

Comment:
A general Julia query, probably not too much to do with this PR specifically. If I keep this line of code, I get a warning on doing `using ChainRules`:
```jl
┌ Info: Precompiling ChainRules [082447d4-558c-5d27-93f4-14fc19e9eca2]
└ @ Base loading.jl:1260
WARNING: Method definition frule(Any, UnionAll, Base.BitArray{N} where N) in module ChainRules at /Users/gaurav/.julia/dev/ChainRulesCore/src/rule_definition_tools.jl:313 overwritten at /Users/gaurav/.julia/dev/ChainRulesCore/src/rule_definition_tools.jl:313.
  ** incremental compilation may be fatally broken for this module **

WARNING: Method definition frule##kw(Any, typeof(ChainRulesCore.frule), Any, UnionAll, Base.BitArray{N} where N) in module ChainRules at /Users/gaurav/.julia/dev/ChainRulesCore/src/rule_definition_tools.jl:313 overwritten at /Users/gaurav/.julia/dev/ChainRulesCore/src/rule_definition_tools.jl:313.
  ** incremental compilation may be fatally broken for this module **

WARNING: Method definition rrule(UnionAll, Base.BitArray{N} where N) in module ChainRules at /Users/gaurav/.julia/dev/ChainRulesCore/src/rule_definition_tools.jl:328 overwritten at /Users/gaurav/.julia/dev/ChainRulesCore/src/rule_definition_tools.jl:328.
  ** incremental compilation may be fatally broken for this module **

WARNING: Method definition rrule##kw(Any, typeof(ChainRulesCore.rrule), UnionAll, Base.BitArray{N} where N) in module ChainRules at /Users/gaurav/.julia/dev/ChainRulesCore/src/rule_definition_tools.jl:328 overwritten at /Users/gaurav/.julia/dev/ChainRulesCore/src/rule_definition_tools.jl:328.
  ** incremental compilation may be fatally broken for this module **
```
A warning, which I believe I understand (no problem here).

And when I remove this line of code (remove re-definition of non-differentiability of `DenseMatrix(::BitArray)`, and keep the below line of code about non-differentiability of `Matrix(::BitArray)` I get the same exact warning:
```jl
┌ Info: Precompiling ChainRules [082447d4-558c-5d27-93f4-14fc19e9eca2]
└ @ Base loading.jl:1260
WARNING: Method definition frule(Any, UnionAll, Base.BitArray{N} where N) in module ChainRules at /Users/gaurav/.julia/dev/ChainRulesCore/src/rule_definition_tools.jl:313 overwritten at /Users/gaurav/.julia/dev/ChainRulesCore/src/rule_definition_tools.jl:313.
  ** incremental compilation may be fatally broken for this module **

WARNING: Method definition frule##kw(Any, typeof(ChainRulesCore.frule), Any, UnionAll, Base.BitArray{N} where N) in module ChainRules at /Users/gaurav/.julia/dev/ChainRulesCore/src/rule_definition_tools.jl:313 overwritten at /Users/gaurav/.julia/dev/ChainRulesCore/src/rule_definition_tools.jl:313.
  ** incremental compilation may be fatally broken for this module **

WARNING: Method definition rrule(UnionAll, Base.BitArray{N} where N) in module ChainRules at /Users/gaurav/.julia/dev/ChainRulesCore/src/rule_definition_tools.jl:328 overwritten at /Users/gaurav/.julia/dev/ChainRulesCore/src/rule_definition_tools.jl:328.
  ** incremental compilation may be fatally broken for this module **

WARNING: Method definition rrule##kw(Any, typeof(ChainRulesCore.rrule), UnionAll, Base.BitArray{N} where N) in module ChainRules at /Users/gaurav/.julia/dev/ChainRulesCore/src/rule_definition_tools.jl:328 overwritten at /Users/gaurav/.julia/dev/ChainRulesCore/src/rule_definition_tools.jl:328.
  ** incremental compilation may be fatally broken for this module **
```

, it didn't even point me to the line that was causing the error (was it `DenseMatrix` line or the `Matrix` line) which often leads me to think over about the non-exactness of the error and warning details in Julia (probably atleast compared to Python), is it just me or am I wrong in thinking like that?

Reply:
That is because of the bug that is fixed by https://github.com/JuliaDiff/ChainRulesCore.jl/pull/243
so if you bump the requirement for ChainRulesCore up to 0.9.19 
it should be fixed (0.9.19 is currently registering)